From 8f974ff00f8744dc779393ea02397a2dd8d05c1b Mon Sep 17 00:00:00 2001
Message-Id: <8f974ff00f8744dc779393ea02397a2dd8d05c1b.1421835210.git.chang-joon.lee@intel.com>
In-Reply-To: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
References: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Tue, 4 Nov 2014 17:46:02 +0000
Subject: [PATCH 43/63] FOR_UPSTREAM [VPG]: drm/i915: More request structure
 tracing

Added the '_complete' trace event which occurs when a request structure is
marked as complete.

Change-Id: Ieefabd5706b86fba4e2e766d4049be01222d3bfb
For: VIZ-4377
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
Reviewed-by: Thomas Daniel <Thomas.Daniel@intel.com>
---
 drivers/gpu/drm/i915/i915_gem.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 481a579..dc55567 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -2809,8 +2809,10 @@ void i915_gem_complete_requests_ring(struct intel_engine_cs *ring,
 		if (req->complete)
 			continue;
 
-		if (i915_seqno_passed(seqno, req->seqno))
+		if (i915_seqno_passed(seqno, req->seqno)) {
 			req->complete = true;
+			trace_i915_gem_request_complete(req);
+		}
 	}
 
 	ring->last_read_seqno = seqno;
-- 
1.7.9.5

